### ISSUE TITLE
rebuild.sh fails on first run

### ISSUE THREAD

**Issue by smustgrave** [thinktandem/drupal-contributions#67]
During lando rebuild I get this consistently 

usermod: no changes
usermod: no changes
/bin/sh: 1: /app/scripts/rebuild.sh: not found
ERROR ==> /bin/sh: 1: /app/scripts/rebuild.sh: not found

**Comment by xurizaemon** [comment]
This is because an earlier step has failed.

Setup:
```
~/Projects$ rm -rf drupal-contributions/
~/Projects$ git clone git@github.com:lando/drupal-contributions.git
Cloning into 'drupal-contributions'...
remote: Enumerating objects: 364, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 364 (delta 0), reused 2 (delta 0), pack-reused 354
Receiving objects: 100% (364/364), 82.28 KiB | 205.00 KiB/s, done.
Resolving deltas: 100% (190/190), done.
~/Projects$ cd drupal-contributions/
```

Problem:
```
~/Projects/drupal-contributions$ lando rebuild -y
Rising anew like a fire phoenix from the ashes! Rebuilding app...
sh: 0: getcwd() failed: No such file or directory
usermod: no changes
usermod: no changes
sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
Could not open input file: /app/scripts/get-drupal.php
ERROR ==> Could not open input file: /app/scripts/get-drupal.php 
Killing drupalcontributions_database_1  ... done
```

**Comment by xurizaemon** [comment]
Based on my testing, it seems like a timing issue or a problem which appears on first run.

I suspect that the pre-rebuild hook is unable to execute before rebuild in a Lando environment which is not yet running.

Run 1:

```
$ rm -rf drupal-contributions ; git clone --branch debug git@github.com:xurizaemon/drupal-contributions.git ; cd drupal-contributions ; lando rebuild -y
Cloning into 'drupal-contributions'...
remote: Enumerating objects: 284, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 284 (delta 15), reused 11 (delta 10), pack-reused 258
Receiving objects: 100% (284/284), 67.75 KiB | 173.00 KiB/s, done.
Resolving deltas: 100% (148/148), done.
Rising anew like a fire phoenix from the ashes! Rebuilding app...
sh: 0: getcwd() failed: No such file or directory
usermod: no changes
usermod: no changes
sh: 0: getcwd() failed: No such file or directory
pre rebuild
sh: 0: getcwd() failed: No such file or directory

sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
Could not open input file: /app/scripts/get-drupal.php
ERROR ==> Could not open input file: /app/scripts/get-drupal.php 
```

Run 2:
```
$ lando rebuild -y
Rising anew like a fire phoenix from the ashes! Rebuilding app...
usermod: no changes
usermod: no changes
pre rebuild
/app
README.md  config  phpunit.xml	scripts  web
removed directory 'web/sites/simpletest/browser_output'
removed directory 'web/sites/simpletest'
removed directory 'web/sites'
removed 'web/.gitignore'
removed directory 'web'
Cloning into 'web'...
```

Full run output: https://gist.github.com/xurizaemon/dfc5d55bddfbed7fc52f2dadafc059d1

Fork branch used here has an additional `pwd` and `ls` inserted to show the state of `/app` for that execution; we see those in run 2. In run 1 those commands don't execute as there is no cwd yet.

**Workaround**: execute `lando rebuild` twice on first run. Not lovable, but works.

We should be able to fix this by moving that setup event to another point in execution, so it works for rebuild AND first run?

@smustgrave if you agree, would you rename issue to something like "rebuild.sh fails on first run"?

**Comment by smustgrave** [comment]
Sure no problem!

**Comment by xurizaemon** [comment]
Oooh, I think this bug is complicated by that if you omit to run `lando stop` before removing the directory to test it doesn't show up (because the containers still exist)? Interesting.

Testing addition of a `pre-start` event.

Noting that the initial docs issue https://github.com/lando/drupal-contributions/issues/1 refers to rebuild not start (_Note that we are using rebuild and not the start command. Rebuild has the events to trigger getting the Drupal source code and installation._) - it feels to me like start is more appropriate here, and that comment doesn't tell me what didn't work about that.

Maybe things have changed and we should give start a try again - I'm concerned that using both will do all that work twice.

EDIT: Oh yeah, that's why I did `if [ ! -d web ] ` in the commit above, lol. Past me is smarter me.

**Comment by xurizaemon** [comment]
OK, try this one:

```
rm -rf drupal-contributions ; git clone --branch 67-first-run-build-fix git@github.com:xurizaemon/drupal-contributions.git ; cd drupal-contributions ; lando rebuild -y
```

Should work on first and subsequent runs. Note that you should stop/destroy your container before wiping the directory :)

Output for first and second run: https://gist.github.com/xurizaemon/f3815ca1779253bebb920c4ccb8d9d73

**Comment by gitressa** [comment]
I also have to run `lando rebuild` twice. Though my errors are different:

```
yarn install v1.22.19
error Directory "/app/web/core" doesn't exist
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command. 
usermod: no changes
usermod: no changes
PHP Fatal error:  Uncaught Error: Class "Symfony\Component\HttpKernel\Kernel" not found in /app/web/vendor/drush/drush/src/Preflight/Preflight.php:209
Stack trace:
[...]
```

**Comment by reynoldsalec** [comment]
Should be resolved by PR #74
